hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -208,6 +209,11 @@
       .sinceVersion("0.11.0")
       .withDocumentation("Table checksum is used to guard against partial writes in HDFS. It is added as the last entry in hoodie.properties and then used to validate while reading table config.");
 
+  public static final ConfigProperty<Boolean> METADATA_TABLE_ENABLE = ConfigProperty
+      .key("hoodie.table.metadata.enable")
+      .defaultValue(HoodieMetadataConfig.ENABLE.defaultValue())
+      .withDocumentation("Enable the internal metadata table which serves table metadata like level file listings.");
```

Comment:
How does the reader path look like? What if MDT is disabled in write config but it was true for reader? Do we need to call out something in the doc that if MDT is disabled in write config then queries will not make use of MDT irrespective of reader MDT config is enabled or not?

Reply:
Sure, we need to let users know it. Just thinking maybe we can  let reader side to notice the config in hoodie.properties:
1. if `METADATA_TABLE_ENABLE` false in hoodie.properties, no matter users set on reader, we will use FileSystemBased
2. if `METADATA_TABLE_ENABLE` true in hoodie.properties, and users set MDT true, then we use MDT Based action?

Reply:
And what if  METADATA_TABLE_ENABLE is not present in the config?

Reply:
3. if `METADATA_TABLE_ENABLE` is absent, we could read based on user's option.

Reply:
Hi @codope just add a new ticket https://issues.apache.org/jira/browse/HUDI-3766 here.
Maybe we can handle this flag for reader side for another PR？
Take this patch as a first step solving the possible out-of-sync issue :)